classes/core/SettingsBuilder.php

```diff
@@ -288,8 +331,19 @@ protected function filterRow(stdClass $row, string|array $columns = ['*']): stdC
         }
 
         $columns = Arr::wrap($columns);
-        foreach ($row as $property) {
-            if (!in_array($property, $columns)) {
+
+        // TODO : Investigate how to handle the camel to snake case issue. related to pkp/pkp-lib#10485
+        $settingColumns = $this->model->getSettings();
+        $columns = collect($columns)
+            ->map(
+                fn (string $column): string => in_array($column, $settingColumns)
+                    ? $column
+                    : Str::snake($column)
+            )
+            ->toArray();
+
+        foreach ($row as $property => $value) {
+            if (!in_array($property, $columns) && isset($row->{$property})) {
```

Comment:
@Vitaliy-1 this change seems required as with current implementation , we do not get the desired outcome with specifying the columns as such 
```php
$announcement = Announcement::find(31, ["title"]);
```
Also throws a lot of warnings . Please take a look .